src/Parser/FrostMage/Modules/Features/WintersChill.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+
+import Module from 'Parser/Core/Module';
+import Enemies from 'Parser/Core/Modules/Enemies';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatNumber } from 'common/format';
+
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+
+class WintersChillTracker extends Module {
+  static dependencies = {
+    enemies: Enemies,
+  };
+  totalWintersChill = 0;
+  inChillIceLance= 0;
+
+  on_byPlayer_cast(event) {
+    if (event.ability.guid !== SPELLS.ICE_LANCE_CAST.id) {
+      return;
+    }
+    const enemy = this.enemies.getEntity(event);
+    if (enemy.hasBuff(SPELLS.WINTERS_CHILL.id, event.timestamp)) {
+      this.inChillIceLance += 1;
+    }
+  }
+  on_applydebuff(event) {
+	  if(event.ability.guid !== SPELLS.WINTERS_CHILL.id) {
+		  return;
+	  }
+	  const enemy = this.enemies.getEntity(event);
+	  if(enemy.hasBuff(SPELLS.WINTERS_CHILL.id)) {
```

Comment:
This feels rather unnecessary - you just applied the Winter's Chill debuff so this condition will most likely always return true.

Reply:
Ill take a second look at it, but Winter's Chill has a very very short duration, like just barely long enough to land one Instant Cast Ice Lance into it before it goes away, so i think that was to make sure that its still there when the spell hits, but ill double check.

Reply:
The point is though, if you get to this point, that means you have _just_ applied the Winter's Chill debuff, and while executing this function the time doesn't change. Makes sense in different events maybe, but when you're in `on_byPlayer_applydebuff(event)` and you check if the ability is Winter's Chill, then this will always return true.

Also, you should check the buff's presence based on specific timestamp, so `hasBuff(spellId, event.timestamp)`.

Reply:
Sharrq, you should be checking damage events, check the damage was from Ice Lance and that the target has Winter's Chill

Reply:
Actually i took another look at it, i think that second section was just to count the number of times Winter's Chill was applied and the top one was to count the number of Ice Lances that hit during winter's chill. I believe that is why there is a second one there ... that way i can figure out the number of times they didnt get an ice lance into winter's chill (for the suggestion). Is there a simpler way of getting the number of times a spell was cast or a buff was applied without manually counting it every time it is cast/applied?

Reply:
You can query the Ability Tracker to get casts of certain ability (by adding `abilityTracker: AbilityTracker,` into dependencies and then in suggestion probably do `const iceLanceCasts = this.abilityTracker.getAbility(SPELLS.ICE_LANCE.id).casts || 0;`), but that would count all Ice Lances for example, not just the ones in/outside of the buff.

Reply:
I would need to get a count of the times Winter's Chill was applied to the target, not the number of times i cast Ice Lance. Can that be done with abilityTracker also?

Though I suppose i could just use the Brain Freeze Proc count from BrainFreeze.js since every Brain Freeze buffed Flurry will cause Winter's Chill.

Reply:
No, Ability Tracker only counts casts. You could list BrainFreeze as a dependency but I don't think it's that big of a deal to do 
```jsx
wintersChillCount = 0;

on_byPlayer_applydebuff(event) {
  if (event.ability.guid !== SPELLS.WINTERS_CHILL.id) {
    return;
  }
  this.wintersChillCount += 1;
}
```

Reply:
Ok, i guess i was confused as to what was un-necessary and needed to be removed. I thought it was the on_applydebuff section, but your suggestion is almost identical to what is already in there ... so that must not be what was un-necessary. 

Or did you mean the Winter's Chill part of the on_byPlayer_cast section? If that is what you meant, then i think you may have assumed that Ice Lance is what caused the debuff therefore checking for the debuff is un-necessary ... but it is Flurry that debuff's the target with Winter's Chill..... So essentially, you get a Brain Freeze Proc, you expend that Proc on a free instant cast Flurry, Flurry debuff's the target with Winter's Chill, and you Cast Ice Lance during that Winter's Chill window. That is why im checking to see if Winter's Chill is there when Ice lance is cast/does damage (ill change it to the on damage one per kfinch's suggestion).